lib/wordmove/list.rb

```diff
@@ -0,0 +1,70 @@
+module Wordmove
+  class List
+    class << self
+      def start(cli_options)
+        new(cli_options).start
+      end
+    end
+
+    def initialize(options)
+      @logger = Logger.new(STDOUT).tap { |l| l.level = Logger::INFO }
+      @movefile = Wordmove::Movefile.new(options[:config])
+      @remote_vhosts = []
+      @local_vhost = []
+    end
+
+    def start
+      contents = parse_movefile(movefile: movefile)
+      generate_vhost_list(contents: contents)
+      output
+    end
+
+    def output_string(vhost_list:)
+      return 'vhost list is empty' if vhost_list.empty?
+
+      ''.tap do |retval|
+        vhost_list.each do |entry|
+          retval << "#{entry[:env]}: #{entry[:vhost]}\n"
+        end
+      end
+    end
```

Comment:
> but personally I found it a bit difficult read.

Thanks! I totally agree with this point of view.
Above is so nice for me and like a charm ❤️ 

Sometime I use the tap to prevent the deep nest.
(Exp. NestedIterators
https://github.com/troessner/reek/blob/master/docs/Nested-Iterators.md)